code/app/src/main/java/com/kernelcrew/moodapp/ui/OtherUserProfile.java

```diff
@@ -77,6 +94,69 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
             emailText.setText("");
         }
 
+        // Initialize and wire up the Follow button
+        followButton = view.findViewById(R.id.followButton);
+
+        // If it's your own profile, hide or disable the follow button
+        if (currentUser != null && uidToLoad != null && uidToLoad.equals(currentUser.getUid())) {
+            followButton.setVisibility(View.GONE);
+        } else {
+            // Otherwise, check if you've already sent a follow request
+            if (currentUser != null && uidToLoad != null) {
+                FirebaseFirestore db = FirebaseFirestore.getInstance();
+                db.collection("users")
+                        .document(uidToLoad)
+                        .collection("followRequests")
+                        .document(currentUser.getUid())
+                        .get()
+                        .addOnSuccessListener(doc -> {
+                            if (doc.exists()) {
+                                // Already requested
+                                followButton.setEnabled(false);
+                                followButton.setText("Requested");
+                            } else {
+                                // Not requested yet
+                                followButton.setEnabled(true);
+                                followButton.setText("Follow");
+                            }
+                        })
+                        .addOnFailureListener(e ->
+                                Log.e(TAG, "Failed to check followRequests doc", e)
+                        );
+            }
+
+            followButton.setOnClickListener(v -> sendFollowRequest());
+        }
+
         return view;
     }
+
+    /**
+     * Sends a follow request from the current user to the user with uidToLoad.
+     */
+    private void sendFollowRequest() {
+        FirebaseUser currentUser = FirebaseAuth.getInstance().getCurrentUser();
+        if (currentUser == null) {
+            Log.e(TAG, "Current user is null; cannot send follow request.");
+            return;
+        }
+        if (uidToLoad == null) {
+            Log.e(TAG, "Target user ID is null; cannot send follow request.");
+            return;
+        }
+
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
```

Comment:
All db interactions should be placed in a provider to simplify testing. Ie. it should be possible to do certain things without even pressing buttons in the UI -- that will save some time and avoid flakiness by letting tests stay more focused.